src/features/users/usersSlice.ts

```diff
@@ -0,0 +1,47 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+
+export interface CounterUsersState {
+  users: User[];
+  selectedUser: User | null;
+}
+
+const initialValue: CounterUsersState = {
+  users: [],
+  selectedUser: null,
+};
+
+export const fetchUsers = createAsyncThunk('users/fetchUsers', async () => {
+  const response = await getUsers();
+
+  return response;
+});
```

Comment:
Consider adding error handling for the `fetchUsers` thunk to manage potential failures during the user fetching process. This can help in providing feedback to the user and debugging.